### ISSUE TITLE
NC13 Users Personal Information Does Not Save

### ISSUE THREAD

**Issue by DanLeonelli** [nextcloud/server#8409]
### Steps to reproduce
1. Go to user settings
2. Try to write your email address or any other detail
3. Go back there: it never saved

### Expected behaviour
Information should auto save

### Actual behaviour
It doesn't save

### Server configuration

**Operating system**:

**Web server:**

**Database:**

**PHP version:**

**Nextcloud version:** (see Nextcloud admin page) 13.0.0

**Updated from an older Nextcloud/ownCloud or fresh install:** Updated

**Where did you install Nextcloud from:** Web

**Signing status:**
<details>
<summary>Signing status</summary>

```
Login as admin user into your Nextcloud and access 
http://example.com/index.php/settings/integrity/failed 
paste the results here.
```
</details>

**List of activated apps:**
<details>
<summary>App list</summary>

```
If you have access to your command line run e.g.:
sudo -u www-data php occ app:list
from within your Nextcloud installation folder
```
</details>

**Nextcloud configuration:**
<details>
<summary>Config report</summary>

```
If you have access to your command line run e.g.:
sudo -u www-data php occ config:list system
from within your Nextcloud installation folder

or 

Insert your config.php content here. 
Make sure to remove all sensitive content such as passwords. (e.g. database password, passwordsalt, secret, smtp password, …)
```
</details>

**Are you using external storage, if yes which one:** local/smb/sftp/...

**Are you using encryption:** yes/no

**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...

#### LDAP configuration (delete this part if not used)
<details>
<summary>LDAP config</summary>

```
With access to your command line run e.g.:
sudo -u www-data php occ ldap:show-config
from within your Nextcloud installation folder

Without access to your command line download the data/owncloud.db to your local
computer or access your SQL server remotely and run the select query:
SELECT * FROM `oc_appconfig` WHERE `appid` = 'user_ldap';


Eventually replace sensitive data as the name/IP-address of your LDAP server or groups.
```
</details>

### Client configuration
**Browser:**

**Operating system:**

#### Web server error log
<details>
<summary>Web server error log</summary>

```
Insert your webserver log here
```
</details>

#### Nextcloud log (data/nextcloud.log)
<details>
<summary>Nextcloud log</summary>

```
Insert your Nextcloud log here
```
</details>

#### Browser log
<details>
<summary>Browser log</summary>

```
Insert your browser log here, this could for example include:

a) The javascript console log
b) The network log
c) ...
```
</details>

I really wish I could go back to version 12. This one has brought me nothing but issues and the worst thing is that no one ever answers them.

**Comment by stratege1401** [comment]
each data input in the "box" need to be validated by enter ... 
Also, it could be a browser cache problem.
try to re-input data
then close browser
then clear cahe
then reconnect

**Comment by DanLeonelli** [comment]
@stratege1401 Already tried that and it doesn't work. No matter how many times you press enter, the only fields that save are: Name and phone number, nothing else will. Already tried in different browsers and in different laptops. It just won't save after upgrading to NC13.

**Comment by stratege1401** [comment]
did you check permission on your www/ directory ...
usually www-data:www-data with 755.

do you have access to the nextcloud.log and the DB log ?
on my debian install my nextcloud.log is located in my /mnt/.../nextcloud.log
and /var/log/mysql/error.log

best thing to do is clear both logs and retry in order to have real info on whats is going on

**Comment by d0tfile** [comment]
@stratege1401, as i see not only by enter, changing focus works too.

@DanLeonelli, maybe something wrong with your setup? Did you check browser and webserver logs? I have no issue with Nextcloud 13 (nginx 1.12.2 + php-fpm 7.2 on Ubuntu 16.04).

**Comment by DanLeonelli** [comment]
@d0tfile No error information about this on the logs. I upgraded many days ago, but I've just started getting reports that email address won't save, not with existing users and I tried creating new users and it just won't save. Also the privacy on each box won't save unless I try like 4 or 5 times. I really don't know what to do as people won't be able to get email notifications if their email address can't be saved.

**Comment by stratege1401** [comment]
`@d0tfile No error information about this on the logs. I upgraded many days ago, but I've just started getting reports that email address won't save, not with existing users and I tried creating new users and it just won't save. Also the privacy on each box won't save unless I try like 4 or 5 times. I really don't know what to do as people won't be able to get email notifications if their email address can't be saved.`

could you copy/past your nextcloud.log and your db error log

**Comment by DanLeonelli** [comment]
@stratege1401 The permission is set to 755, as always. It used to work perfectly fine before update.

I have no idea where to find the nextcloud.log or the db error log. All I can see is just the "Logging" section as an admin:

<img width="1407" alt="screen shot 2018-02-17 at 2 13 11 pm" src="https://user-images.githubusercontent.com/35517527/36344745-bb9c2610-13ec-11e8-8649-1a6d89486bcc.png">

Error | PHP | idn_to_ascii() expects parameter 3 to be integer, string given at /home1/tlsoluti/public_html/mi/lib/private/Mail/Mailer.php#212 | 2018-02-17T14:10:04-0500
-- | -- | -- | --
Error | PHP | Use of undefined constant INTL_IDNA_VARIANT_UTS46 - assumed 'INTL_IDNA_VARIANT_UTS46' at /home1/tlsoluti/public_html/mi/lib/private/Mail/Mailer.php#212 | 2018-02-17T14:10:04-0500
Error | PHP | idn_to_ascii() expects parameter 3 to be integer, string given at /home1/tlsoluti/public_html/mi/lib/private/Mail/Mailer.php#212 | 2018-02-17T14:10:04-0500
Error | PHP | Use of undefined constant INTL_IDNA_VARIANT_UTS46 - assumed 'INTL_IDNA_VARIANT_UTS46' at /home1/tlsoluti/public_html/mi/lib/private/Mail/Mailer.php#212 | 2018-02-17T14:10:04-0500
Error | PHP | idn_to_ascii() expects parameter 3 to be integer, string given at /home1/tlsoluti/public_html/mi/lib/private/Mail/Mailer.php#212 | 2018-02-17T14:10:04-0500
Error | PHP | Use of undefined constant INTL_IDNA_VARIANT_UTS46 - assumed 'INTL_IDNA_VARIANT_UTS46' at /home1/tlsoluti/public_html/mi/lib/private/Mail/Mailer.php#212 | 2018-02-17T14:10:04-0500
Error | PHP | idn_to_ascii() expects parameter 3 to be integer, string given at /home1/tlsoluti/public_html/mi/lib/private/Mail/Mailer.php#212 | 2018-02-17T14:10:04-0500
Error | PHP | Use of undefined constant INTL_IDNA_VARIANT_UTS46 - assumed 'INTL_IDNA_VARIANT_UTS46' at /home1/tlsoluti/public_html/mi/lib/private/Mail/Mailer.php#212 | 2018-02-17T14:10:04-0500
Error | PHP | idn_to_ascii() expects parameter 3 to be integer, string given at /home1/tlsoluti/public_html/mi/lib/private/Mail/Mailer.php#212 | 2018-02-17T14:10:01-0500
Error | PHP | Use of undefined constant INTL_IDNA_VARIANT_UTS46 - assumed 'INTL_IDNA_VARIANT_UTS46' at /home1/tlsoluti/public_html/mi/lib/private/Mail/Mailer.php#212 | 2018-02-17T14:10:01-0500
Error | PHP | idn_to_ascii() expects parameter 3 to be integer, string given at /home1/tlsoluti/public_html/mi/lib/private/Mail/Mailer.php#212 | 2018-02-17T14:10:01-0500
Error | PHP | Use of undefined constant INTL_IDNA_VARIANT_UTS46 - assumed 'INTL_IDNA_VARIANT_UTS46' at /home1/tlsoluti/public_html/mi/lib/private/Mail/Mailer.php#212 | 2018-02-17T14:10:01-0500
Error | PHP | idn_to_ascii() expects parameter 3 to be integer, string given at /home1/tlsoluti/public_html/mi/lib/private/Mail/Mailer.php#212 | 2018-02-17T14:10:01-0500

**Comment by stratege1401** [comment]
The real log is usually located in:
/var/log/nextcloud.log
or
in the data directory of your data side of nextcloud
for me, my nextcloud conf is located at /var/www/html
my nextclout data in my raid space witch is mounted as /mnt/lvm1/...
.../user1
.../user2
.../...
.../appdata_occo4kfce11o
.../updater-occo4kfce11o
;;;/nextcloud.log

that is the real log !

First thing, you are not the only one to have what i think is a bug.:
https://help.nextcloud.com/t/cannot-add-email-to-profile/27347

For me, there is a problem with .../lib/private/Mail/Mailer.php
I think you have a broken dependency with pecl with must be => 5.4.0/PECL 2.0.0b1

depends of your php-version 5.x or 7.x, you can try to fix by doing
apt-get install php-pear
then apt-get install php5-dev
then apt-get install libpcre3-dev libcurl3-openssl-dev
then pecl install pecl_http
edit nano /etc/php5/apache2/php.ini
and add whatever extention the pecj command give you, usually extension=http.so
create file /etc/php5/apache2/conf.d/pecl-http.ini
add three line:
extension=raphf.so
extension=propro.so
extension=http.so
restart apache

**Comment by DanLeonelli** [comment]
@stratege1401 I have a shared hosting. I have no idea on how to do everything that you've mentioned. It's like reading Chinese :(

**Comment by stratege1401** [comment]
sorry, just editing the message below.

I case of share hosting, ask your admin or your coloc company

Your linux version is missing dependency. I guess tou need to have the package liste above for your system to work with nextcloud.

The most important is 5.4.0/PECL 2.0.0b1

If you are not able to update the missing dependency, you might not use nextcloud because your shared hosting wont work.

**Comment by stratege1401** [comment]
i have a bad habit to write/post/edit/edit/edit/ my message, so you might read v3 of my message, when i have just finish editing v4 ... sorry for that

**Comment by stratege1401** [comment]
ask to have your system update to 


Release 3.0.0: | PHP Version: PHP 5.2.4 or newerPEAR Package: PEAR 1.4.8 or newer to fix your system.

**Comment by stratege1401** [comment]
> 6T22:59:59+00:00","remoteAddr":"190.119.250.33","user":"dsanchez","app":"PHP","method":"POST","url":"/mi/index.php/apps/mail/api/accounts","message":"fwrite(): SSL operation failed with code 1. OpenSSL Error messages:\nerror:1409F07F:SSL routines:SSL3_WRITE_PENDING:bad write retry at /home1/tlsoluti/public_html/mi/apps/mail/vendor/pear-

this is typical of a bad php/pearl upgrade because of bad openssl dependency or because some changes exist in command version 
Check: http://php.net/manual/en/migration56.openssl.php

**Comment by DanLeonelli** [comment]
@stratege1401 Seems like the versions are updated:

Apache | 2.2.34
PHP | 5.6.32
MySQL | 5.6.32-78.1
Perl | 5.24.1

**Comment by stratege1401** [comment]
my apache2 -V gives:

```
Server version: Apache/2.4.29 (Debian)
Server built:   2018-01-14T11:01:58
Server's Module Magic Number: 20120211:68
Server loaded:  APR 1.6.3, APR-UTIL 1.6.1
Compiled using: APR 1.6.3, APR-UTIL 1.6.1
Architecture:   64-bit
Server MPM:
Server compiled with....
 -D APR_HAS_SENDFILE
 -D APR_HAS_MMAP
 -D APR_HAVE_IPV6 (IPv4-mapped addresses enabled)
 -D APR_USE_SYSVSEM_SERIALIZE
 -D APR_USE_PTHREAD_SERIALIZE
 -D SINGLE_LISTEN_UNSERIALIZED_ACCEPT
 -D APR_HAS_OTHER_CHILD
 -D AP_HAVE_RELIABLE_PIPED_LOGS
 -D DYNAMIC_MODULE_LIMIT=256
 -D HTTPD_ROOT="/etc/apache2"
 -D SUEXEC_BIN="/usr/lib/apache2/suexec"
 -D DEFAULT_PIDLOG="/var/run/apache2.pid"
 -D DEFAULT_SCOREBOARD="logs/apache_runtime_status"
 -D DEFAULT_ERRORLOG="logs/error_log"
 -D AP_TYPES_CONFIG_FILE="mime.types"
 -D SERVER_CONFIG_FILE="apache2.conf"
```
your pearl is 5.24.1, must be => 5.4.0 at least. My perl -V gives:

```
perl -v
This is perl 5, version 26, subversion 1 (v5.26.1) built for x86_64-linux-gnu-thread-multi
```


and mysql gives
```
mysql -V
mysql  Ver 15.1 Distrib 10.1.26-MariaDB, for debian-linux-gnu (x86_64) using readline 5.2
```
and to finis, openssl
```
openssl version -a
OpenSSL 1.1.0f  25 May 2017
```

you definitely need tu update

**Comment by stratege1401** [comment]
check: https://docs.nextcloud.com/server/13/admin_manual/installation/source_installation.html

i would say: ask your share host admin to update your system ...

**Comment by blizzz** [comment]
the INTL_IDNA_VARIANT_UTS46-issues are fixed with #9100 otherwise, as @stratege1401 says, it's about underlying systems → closing
